stdlib/public/Concurrency/Actor.cpp

```diff
@@ -538,12 +615,23 @@ static bool swift_task_isCurrentExecutorWithFlagsImpl(
 
 // Check override of executor checking mode.
 static void swift_task_setDefaultExecutorCheckingFlags(void *context) {
-  bool useLegacyMode = swift_bincompat_useLegacyNonCrashingExecutorChecks();
-  auto checkMode = static_cast<swift_task_is_current_executor_flag *>(context);
-  if (!useLegacyMode) {
-    *checkMode = swift_task_is_current_executor_flag(
-        *checkMode | swift_task_is_current_executor_flag::Assert);
-  }
+  SWIFT_TASK_DEBUG_LOG("executor checking: swift_task_setDefaultExecutorCheckingFlags = %d", nullptr);
+  auto useLegacyMode = swift_bincompat_useLegacyNonCrashingExecutorChecks();
+  SWIFT_TASK_DEBUG_LOG("executor checking: use legacy mode = %d", useLegacyMode);
+  auto *options = static_cast<swift_task_is_current_executor_flag *>(context);
+  SWIFT_TASK_DEBUG_LOG("executor checking: options = %d", *options);
+//  if (!useLegacyMode) {
```

Comment:
Presumably should be uncommented or removed.

Reply:
Thanks yeah; let me do some more work here and cleanups as well; trying to orchestrate not conflicting too hard with @al45tair as well